core/peerset/src/lib.rs

```diff
@@ -433,7 +433,7 @@ impl Stream for Peerset {
 				return Ok(Async::Ready(Some(message)));
 			}
 			match try_ready!(self.rx.poll()) {
-				None => return Ok(Async::Ready(None)),
+				None => return Ok(Async::NotReady),
```

Comment:
Since we never return `Async::Ready(None)`, we should not need to return an `Option` at all.

Reply:
You should make you familiar with rust futures.